Added tutorial chapter on coordinates of an mobject (#1210)
ManimCommunity/manim: docs/source/tutorials/building_blocks.rst

	   self.play(ApplyMethod(square.shift, UP), run_time=3)
	   self.wait(1)

Using coordinates of an mobject
===============================

Mobjects contain points that define their boundaries.

> These points can be used to add other mobjects respectively to each other, e.g. by methods like ``m.get_center()`` ``m.get_top()``
> and ``m.get_start()`. Here is an example of some important coordinates:

Author: ```suggestion
These points can be used to add other mobjects respectively to each other, e.g. by methods like ``m.get_center()`` , ``m.get_top()``
and ``m.get_start()``. Here is an example of some important coordinates:
```

	   self.play(ApplyMethod(square.shift, UP), run_time=3)
	   self.wait(1)

> Using coordinates of an mobject
> ===============================

Reviewer: ```suggestion
Using coordinates of a mobject
==============================
```

Mobjects contain points that define their boundaries.

> These points can be used to add other mobjects respectively to each other, e.g. by methods like ``m.get_center()`` , ``m.get_top()``

Reviewer: ```suggestion
These points can be used to add other mobjects respectively to each other, e.g. by methods like :meth:`~.Mobject.get_center` , :meth:`~.Mobject.get_top`
```

> and ``m.get_start()``. Here is an example of some important coordinates:

Reviewer: ```suggestion
and :meth`~.Mobject.get_start`. Here is an example of some important coordinates:
```

            point_start= a.get_start()
            point_end  = a.get_end()
            point_center = a.get_center()
            self.add(Text(f"a.get_start() = {np.round(point_start,2).tolist()}").scale(0.5).to_edge(UR).set_color(YELLOW))
            self.add(Text(f"a.get_end() = {np.round(point_end,2).tolist()}").scale(0.5).next_to(self.mobjects[-1],DOWN).set_color(RED))
            self.add(Text(f"a.get_center() = {np.round(point_center,2).tolist()}").scale(0.5).next_to(self.mobjects[-1],DOWN).set_color(BLUE))

            self.add(Dot(a.get_start()).set_color(YELLOW).scale(2))
            self.add(Dot(a.get_end()).set_color(RED).scale(2))
            self.add(Dot(a.get_top()).set_color(GREEN_A).scale(2))
            self.add(Dot(a.get_bottom()).set_color(GREEN_D).scale(2))
            self.add(Dot(a.get_center()).set_color(BLUE).scale(2))
            self.add(Dot(a.point_from_proportion(0.5)).set_color(ORANGE).scale(2))
            self.add(*[Dot(x) for x in a.get_points()])
            self.add(a)

Transforming mobjects into other mobjects
=========================================

> It is also possible to transform an mobject into another mobject like this:

Reviewer: ```suggestion
It is also possible to transform a mobject into another mobject like this:
```

.. manim:: ExampleTransform

    class ExampleTransform(Scene):
        def construct(self):
            self.camera.background_color = WHITE
            m1 = Square().set_color(RED)
            m2 = Rectangle().set_color(RED).rotate(0.2)
            self.play(Transform(m1,m2))

The Transform function maps points of the previous mobject to the points of the next mobject.
This might result in strange behaviour, e.g. when the dots of one mobject are arranged clockwise and the other points are arranged counterclockwise.
Here it might help to use the `flip` function and reposition the points via the  `roll` function of numpy:

.. manim:: ExampleRotation

    class ExampleRotation(Scene):

>         def construct(self):

Author: This example needs to be updated, as the setting of points was changed to counterclockwise

Reviewer: Any updates on this example now that the default is counterclockwise (CCW) instead of CW?

Author: I updated the example, so it still illustrates what happens when points are rolled with `roll`

>             self.camera.background_color = WHITE
>             m1a = Square().set_color(RED).shift(LEFT)
>             m1b = Circle().set_color(RED).shift(LEFT)
>             m2a= Square().set_color(BLUE).shift(RIGHT)
>             m2b= Circle().set_color(BLUE).shift(RIGHT)
>             m3a= Square().set_color(GREEN).shift(3*RIGHT)
>             m3b= Circle().set_color(GREEN).shift(3*RIGHT)
> 
>             points = m2a.points
>             points = np.flip(points, axis=0)
>             points = np.roll(points, int(len(points)/4), axis=0)
>             m2a.points = points
> 
>             points = m3a.points
>             points = np.flip(points, axis=0)
>             points = np.roll(points, int(len(points)/4)*2, axis=0)
>             m3a.points = points
> 
>             self.play(Transform(m1a,m1b),Transform(m2a,m2b),Transform(m3a,m3b), run_time=1)

Author: ```suggestion
            self.camera.background_color = WHITE
            m1a = Square().set_color(RED).shift(LEFT)
            m1b = Circle().set_color(RED).shift(LEFT)
            m2a= Square().set_color(BLUE).shift(RIGHT)
            m2b= Circle().set_color(BLUE).shift(RIGHT)
            m3a= Square().set_color(GREEN).shift(3*RIGHT)
            m3b= Circle().set_color(GREEN).shift(3*RIGHT)
            
            points = m2a.points
            points = np.roll(points, int(len(points)/4), axis=0)
            m2a.points = points
            
            self.play(Transform(m1a,m1b),Transform(m2a,m2b), run_time=1)
```
